src/components/ForgetPassword.jsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState,useEffect } from 'react';
+import { useForm } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { Typography } from '@mui/material';
+import { ToastContainer } from 'react-toastify';
+import Loading from './Loading';
+import { requestPasswordReset,reset } from '../states/features/auth/authSlice';
+
+import { ErrorNotification, successNotification } from './Notification';
+
+
+
+const ForgetPassword = () => {
+  const [email, setEmail] = useState('');
+  
+  const { register, handleSubmit, } = useForm();
+  const navigate=useNavigate()
+
+  const handleEmailChange = (e) => {
+    setEmail(e.target.value);
+  };
+
+  const dispatch = useDispatch();
+  const { isLoading,isError,isSuccess, message } = useSelector((state) => {
+    return state.auth;
+  });
+
+  
+
+
+const onSubmit = () => {
+  if (!email) {
+    message('Please enter your email address.');
+  } else {
+    dispatch(requestPasswordReset(email))
+    if (isSuccess) {
+      successNotification("Email is Successfully Sent!")
+      
+      navigate('/login');
+    } else {
+      ErrorNotification("Email is not found")
+      
+    }
```

Comment:
These codes should not be placed here because you cannot submit the form and then track the success of the submission within the same function. They should be placed in a useEffect hook instead. Please consider revising your code accordingly. In fact it was done below. Therefore remove those codes from the submit function